staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -168,12 +172,13 @@ var internalPackages = []string{"client-go/tools/cache/"}
 // objects and subsequent deltas.
 // Run will exit when stopCh is closed.
 func (r *Reflector) Run(stopCh <-chan struct{}) {
-	klog.V(3).Infof("Starting reflector %v (%s) from %s", r.expectedTypeName, r.resyncPeriod, r.name)
-	wait.Until(func() {
+	klog.V(2).Infof("Starting reflector %s (%s) from %s", r.expectedTypeName, r.resyncPeriod, r.name)
```

Comment:
Was it intentional that the log level got bumped up from 3 to 2?

I just updated to client-go from 1.21.1 and these messages started to pop up quite a lot in my client (an e2e test suite).

Reply:
See https://github.com/kubernetes/kubernetes/pull/102788

Reply:
https://github.com/kubernetes/kubernetes/pull/87795#discussion_r376224120

I guess that was my intention. In practice we found reflector starting is an important debugging point and it only prints once when reflector initially created so won’t be too verbose.

Reply:
Yes. I'm running an e2e.test suite where essentially each test suite is independent and therefore instantiates its own reflectors.

Reply:
Would this log help debug then? I.e. IMO it makes sense for v2 log or even v1 log to include critical time points such as component start / finish etc. in general.
@wojtek-t thoughts?

Reply:
> Would this log help debug then?

No. For me it is just additional noise. That's why I investigated.

> I.e. IMO it makes sense for v2 log or even v1 log to include critical time points such as component start / finish etc. in general.

But client-go as a library cannot make assumptions about what constitutes component start/finish. In particular one cannot assume that this code is only called at the start and end of a program run. E2E testing is one counter example, but there might very well be others.
 
A library that gets embedded in command line tools should be very cautious about logging at a level that is visible by default. I would not expect any output from client-go at V < 3 unless that output is relevant for the end-user of that command line tool.

Reply:
> I don't think it really was - we should fix that and backport.

Here's the backport: https://github.com/kubernetes/kubernetes/pull/102800

Reply:
> Would this log help debug then? I.e. IMO it makes sense for v2 log or even v1 log to include critical time points such as component start / finish etc. in general.
@wojtek-t thoughts?

Reflector starts may happen not only on component start/finish. Any time watch is not able to resume it may also happen. If you really want those logs, you can always configure verbosity per file. I don't think it should be v=2 by default.